admin/data/sql/migrations/20191126095851.do.add-check-status-calc-func.sql

```diff
@@ -0,0 +1,117 @@
+IF OBJECT_ID (N'dbo.ufnCalcCheckStatusID', N'FN') IS NOT NULL
+    DROP FUNCTION ufnCalcCheckStatusID;
+GO
+
+CREATE FUNCTION dbo.ufnCalcCheckStatusID (@checkId int)
+    RETURNS int
+AS
+BEGIN
+    DECLARE
+        @checkTimeLimit tinyint,
+        @complete bit,
+        @isLiveCheck bit,
+        @pinExpiresAt datetimeoffset,
+        @processingFailed bit,
+        @pupilLoginDate datetimeoffset,
+        @receivedByServerAt datetimeoffset,
+        @ret int,
+        @startedAt datetimeoffset
+    ;
+
+    SELECT
+    TOP 1
+    @checkTimeLimit = checkTimeLimit
+    FROM
+        [mtc_admin].[settings];
+
+    SELECT
+            @complete = complete,
+            @isLiveCheck = isLiveCheck,
+            @ret = checkStatus_id, -- set the return val to the current status
+            @pinExpiresAt = pinExpiresAt,
+            @processingFailed = processingFailed,
+            @pupilLoginDate = pupilLoginDate,
+            @receivedByServerAt = receivedByServerAt,
+            @startedAt = startedAt
+    FROM
+        [mtc_admin].[check] c LEFT OUTER JOIN
+        [mtc_admin].[checkPin] cp ON (c.id = cp.check_id)
+    WHERE
+            c.id = @checkId;
+
+    -- NEW
+    IF (@pupilLoginDate IS NULL AND
+        @startedAt IS NULL AND
```

Comment:
@GuyHarwood - given the imminancy of the new world so we want to remove these startedAt vars?

Reply:
yes, makes sense to me

Reply:
also the enum entry, which ive commented on separately

Reply:
Okay- let's queue these up for the next PR as otherwise, we will break the build with this PR.  Is that okay with you @GuyHarwood?

The next PR will move the views and other consumers of the pupil state over to the new pupil state fields so we can keep the build green.

Reply:
yes, fine with me.